cpp/open3d/t/io/file_format/FileSPLAT.cpp

```diff
@@ -0,0 +1,354 @@
+
+    // Total Gaussians
+    long num_gaussians =
+            static_cast<long>(pointcloud.GetPointPositions().GetLength());
+
+    // Open splat file
+    splat_file = fopen(filename.c_str(), "wb");
+    if (!splat_file) {
+        utility::LogWarning("Write SPLAT failed: unable to open file: {}.",
+                            filename);
+        return false;
+    }
+
+    // Write to SPLAT
+    utility::CountingProgressReporter reporter(params.update_progress);
+    reporter.SetTotal(num_gaussians);
+
+    for (int64_t i = 0; i < num_gaussians; i++) {
+        // Positions
+        DISPATCH_DTYPE_TO_TEMPLATE(positions_attr.dtype_, [&]() {
+            const scalar_t *positions_ptr =
+                    static_cast<const scalar_t *>(positions_attr.data_ptr_);
+            for (int idx_offset = positions_attr.group_size_ * i;
+                 idx_offset < positions_attr.group_size_ * (i + 1);
+                 ++idx_offset) {
+                splat_write_float(splat_file, positions_ptr[idx_offset]);
+            }
+        });
+
+        // Scale
+        DISPATCH_DTYPE_TO_TEMPLATE(scale_attr.dtype_, [&]() {
+            const scalar_t *scale_ptr =
+                    static_cast<const scalar_t *>(scale_attr.data_ptr_);
+            for (int idx_offset = scale_attr.group_size_ * i;
+                 idx_offset < scale_attr.group_size_ * (i + 1); ++idx_offset) {
+                splat_write_float(splat_file, scale_ptr[idx_offset]);
+            }
+        });
+
+        // Color
+        DISPATCH_DTYPE_TO_TEMPLATE(opacity_attr.dtype_, [&]() {
+            int f_dc_offset = f_dc_attr.group_size_ * i;
+            int opacity_offset = opacity_attr.group_size_ * i;
+            const scalar_t *f_dc_ptr =
+                    static_cast<const scalar_t *>(f_dc_attr.data_ptr_);
+            const scalar_t *opacity_ptr =
+                    static_cast<const scalar_t *>(opacity_attr.data_ptr_);
+            Eigen::Vector4i color = ComputeColor(f_dc_ptr + f_dc_offset,
+                                                 opacity_ptr + opacity_offset);
+
+            for (int idx = 0; idx < 4; ++idx) {
+                splat_write_byte(splat_file, int_to_byte(color[idx]));
+            }
+        });
+
+        // Rot
+        DISPATCH_DTYPE_TO_TEMPLATE(rot_attr.dtype_, [&]() {
+            Eigen::Vector4f rot;
+            const scalar_t *rot_ptr =
+                    static_cast<const scalar_t *>(rot_attr.data_ptr_);
+            int rot_offset = rot_attr.group_size_ * i;
+
+            rot << rot_ptr[rot_offset], rot_ptr[rot_offset + 1],
+                    rot_ptr[rot_offset + 2], rot_ptr[rot_offset + 3];
+
+            rot = (((rot / rot.norm()) * 128.0) +
+                   Eigen::Vector4f::Constant(128.0));
+            Eigen::Vector4i int_rot =
+                    rot.cwiseMin(255.0).cwiseMax(0.0).cast<int>();
+
+            for (int idx = 0; idx < 4; ++idx) {
+                splat_write_byte(splat_file, int_to_byte(int_rot[idx]));
```

Comment:
The bounds checking inside int_to_byte is unnecessary, since we have clamped the values above. Simpler `static_cast` should be faster.

Reply:
Ok. Done.